pandas/core/indexing.py

```diff
@@ -2215,8 +2233,6 @@ def _getitem_axis(self, key, axis=None):
 
         # a single integer
         else:
-            key = self._convert_scalar_indexer(key, axis)
```

Comment:
Can you check if there is any remaining occasion where `_convert_scalar_indexer` is used for `iloc` (in principle it shouldn't?). Because if not, `_convert_scalar_indexer` has some redundant code

Reply:
Yes, there is, and it was related to a commit I pushed but then removed, because unfortunately it broke some tests and I don't have time to check them in the next days. So I went for the obvious change only, will add the rest in some other PR (which I think will be quite large, as I will also touch code in indexes).

Reply:
OK. Can you point to where it is currently used? (to have this on record) 
Is it worth to add a todo comment there?

Reply:
I'll rebase my other PR on this once merged, push it and open an issue referring to it